bpython/simpleeval.py

```diff
@@ -0,0 +1,145 @@
+# encoding: utf-8
+"""simple evaluation of side-effect free code
+
+In order to provide fancy completion, some code can be executed safely.
+
+"""
+
+from ast import *
+from six import string_types
+
+from bpython import line as line_properties
+
+class EvaluationError(Exception):
+    """Raised if an exception occurred in safe_eval."""
+
+
+def safe_eval(expr, namespace):
+    """Not all that safe, just catches some errors"""
+    try:
+        return eval(expr, namespace)
+    except (NameError, AttributeError, SyntaxError):
+        # If debugging safe_eval, raise this!
+        # raise
+        raise EvaluationError
+
+def simple_eval(node_or_string, namespace=None):
+    """
+    Safely evaluate an expression node or a string containing a Python
+    expression.  The string or node provided may only consist of:
+    * the following Python literal structures: strings, numbers, tuples,
+        lists, dicts, booleans, and None.
+    * variable names causing lookups in the passed in namespace or builtins
+    * getitem calls using the [] syntax on objects of the types above
+    * getitem calls on subclasses of the above types if they do not override
+        the __getitem__ method and do not override __getattr__ or __getattribute__
+        (or maybe we'll try to clean those up?)
```

Comment:
Add tests for exactly these capabilities - I didn't realize complex numbers broke in Python 3 because there weren't tests for it.